src/components/structures/RoomView.js

```diff
@@ -496,64 +493,72 @@ module.exports = React.createClass({
             this.refs.searchResultsPanel.resetScrollState();
         }
 
-        this.nextSearchBatch = null;
-        this._getSearchBatch(term, scope);
+        // make sure that we don't end up showing results from
+        // an aborted search by keeping a unique id.
+        //
+        // todo: should cancel any previous search requests.
+        this.searchId = new Date().getTime();
+
+        var filter;
+        if (scope === "Room") {
+            filter = {
+                // XXX: it's unintuitive that the filter for searching doesn't have the same shape as the v2 filter API :(
+                rooms: [
+                    this.props.roomId
+                ]
+            };
+        }
+
+        if (DEBUG_SCROLL) console.log("sending search request");
+
+        var searchPromise = MatrixClientPeg.get().searchRoomEvents(
+            { filter: filter,
+              term: term,
+            });
+        this._handleSearchResult(searchPromise)
+            .done();
     },
 
-    // fire off a request for a batch of search results
-    _getSearchBatch: function(term, scope) {
-        this.setState({
-            searchInProgress: true,
-        });
+    _backPaginateSearch: function() {
+        if (DEBUG_SCROLL) console.log("sending search back-paginate request");
 
-        // make sure that we don't end up merging results from
-        // different searches by keeping a unique id.
-        //
-        // todo: should cancel any previous search requests.
-        var searchId = this.searchId = new Date().getTime();
+        var searchPromise = MatrixClientPeg.get().backPaginateRoomEventsSearch(
+            this.state.searchResults);
+        this._handleSearchResult(searchPromise)
+            .done();
+    },
 
+    _handleSearchResult: function(searchPromise) {
         var self = this;
+        var searchId = this.searchId;
```

Comment:
Nit: I would prefer if this were named `searchIdProcessed` or something to differentiate it from `this.searchId`. We're keeping a ref here for use in `:541`, and on that line it isn't clear why on earth we're doing `self.searchId != searchId`. However, `self.searchId != searchIdProcessed` makes it clearer why this check is being performed.